test/install_script/nvm_source

```diff
@@ -9,16 +9,20 @@ die () { echo "$@" ; cleanup ; exit 1; }
 NVM_ENV=testing \. ../../install.sh
 
 # nvm_source with no parameter returns the git endpoint
-echo $(nvm_source) | grep "nvm.git$" > /dev/null || die "nvm_source without arguments should return the location of the git repo"
+echo $(nvm_source) | grep "https://github.com/nvm-sh/nvm.git$" > /dev/null || die "nvm_source without arguments should return the location of the git repo"
+NVM_GITHUB_USER="other" echo $(nvm_source) | grep "https://github.com/other/nvm.git$" > /dev/null || die "nvm_source without arguments should return the location of the git repo"
 
 # nvm_source with git parameter returns the location of the nvm repo
-echo $(nvm_source "git") | grep "nvm.git$" > /dev/null || die "nvm_source without arguments should return the location of the git repo"
+echo $(nvm_source "git") | grep "https://github.com/nvm-sh/nvm.git$" > /dev/null || die "nvm_source without arguments should return the location of the git repo"
+NVM_GITHUB_USER="other" echo $(nvm_source "git") | grep "https://github.com/other/nvm.git$" > /dev/null || die "nvm_source without arguments should return the location of the git repo"
 
 # nvm_source with script parameter returns the location of nvm.sh
-echo $(nvm_source "script") | grep "nvm.sh$" > /dev/null || die "nvm_source \"script\" should return the location of nvm.sh"
+echo $(nvm_source "script") | grep "https://raw.githubusercontent.com/nvm-sh/nvm/$(nvm_latest_version)/nvm.sh$" > /dev/null || die "nvm_source \"script\" should return the location of nvm.sh"
+NVM_GITHUB_USER="other" NVM_INSTALL_VERSION="master" echo $(nvm_source "script") | grep "https://raw.githubusercontent.com/other/nvm/master/nvm.sh$" > /dev/null || die "nvm_source \"script\" should return the location of nvm.sh"
 
 # nvm_source with script-nvm-exec parameter returns the location of nvm-exec
-echo $(nvm_source "script-nvm-exec") | grep "nvm-exec$" > /dev/null || die "nvm_source \"script-nvm-exec\" should return the location of nvm.sh"
+echo $(nvm_source "script-nvm-exec") | grep "https://raw.githubusercontent.com/nvm-sh/nvm/$(nvm_latest_version)/nvm-exec$" > /dev/null || die "nvm_source \"script-nvm-exec\" should return the location of nvm.sh"
+NVM_GITHUB_USER="other" NVM_INSTALL_VERSION="master" echo $(nvm_source "script-nvm-exec") | grep "https://raw.githubusercontent.com/other/nvm/master/nvm-exec$" > /dev/null || die "nvm_source \"script-nvm-exec\" should return the location of nvm.sh"
```

Comment:
hmm - this does concern me, since people already try to automate always installing from master, and this change makes it easier. I break `master` all the time, and outside of nvm development, only tagged version numbers should be installed. Not sure there's an alternative here tho.

Reply:
Indeed I dont see much alternative. At least in the test I could put something else than master not to give this idea to other people to do that.
Moreover, this option is not documented, so from outside, people should not use another version that the one given by nvm_latest_version.
We could otherwise check that the version given is an existing tag from the git repo (or an hard coded list taken from that list, for the case of installation by script method).

Reply:
That brings me to a question though, why would you break master so often :D ?

Reply:
Ideally I never do :-) but the standard git workflow is that the default branch is not safe to rely on, and I want to ensure I train people to only install tagged versions.

Reply:
Ok